### PR TITLE
[mediaset] Update extractor for 2018

### PR DESCRIPTION
**Pull request by timendum**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Bug fix

---

### Description of your *pull request* and other information

Mediaset has updated the website and newer videos now have a different id on the CDN, this change update the code to reflect the changes.

### CODE REVIEW

File: youtube_dl/extractor/mediaset.py [hunk 1]
```diff
@@ -70,10 +86,21 @@ def _extract_urls(webpage):
...
         video_id = self._match_id(url)
 
+        media_info = self._download_json(
+            'https://www.video.mediaset.it/html/metainfo.sjson',
+            video_id, 'Downloading media info', query={
+                'id': video_id
+            })
+
+        if media_info.get('video') and media_info['video'].get('guid'):
+            media_id = media_info['video']['guid']
```
Comment (dstftw) [inline youtube_dl/extractor/mediaset.py]: `try_get`.

**Comment by timendum** [comment]
I've reviewed the code, it's cleaner now. Thanks for the comment.

File: youtube_dl/extractor/mediaset.py [hunk 2]
```diff
@@ -70,10 +86,18 @@ def _extract_urls(webpage):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
+        media_info = self._download_json(
+            'https://www.video.mediaset.it/html/metainfo.sjson',
+            video_id, 'Downloading media info', query={
+                'id': video_id
+            })
```
Comment (dstftw) [inline youtube_dl/extractor/mediaset.py]: metainfo is downloaded twice.

**Comment by timendum** [comment]
I've removed the second download of the same info and added a try/except block to go on in case of error in downloading ISM manifest.

**Pull request closed without merging**